src/EventListener/DeserializeListener.php

```diff
@@ -39,28 +38,27 @@ final class DeserializeListener
 
     private $serializer;
     private $serializerContextBuilder;
-    private $formats = [];
+    private $formats;
     private $formatsProvider;
-    private $formatMatcher;
 
     /**
-     * @throws InvalidArgumentException
+     * @param ResourceMetadataFactoryInterface|FormatsProviderInterface|array $resourceMetadataFactory
      */
-    public function __construct(SerializerInterface $serializer, SerializerContextBuilderInterface $serializerContextBuilder, /* FormatsProviderInterface */$formatsProvider, ResourceMetadataFactoryInterface $resourceMetadataFactory = null)
+    public function __construct(SerializerInterface $serializer, SerializerContextBuilderInterface $serializerContextBuilder, $resourceMetadataFactory, ResourceMetadataFactoryInterface $legacyResourceMetadataFactory = null)
```

Comment:
how is this not a BC break?

Reply:
Why would it be? It's even covered by a test.

Reply:
correct me if i'm wrong, but `$formatsProvider` was the 3rd parameter, which was deprecated in 2.3 (line 54 below), but `$formatsProvider` has been removed. can't you only remove it in 3.0?

Reply:
ah i see now, you've just lost the deprecation warning for usage or `FormatsProviderInterface`? it needs to remain?

Reply:
it was 2.3, but now the warning says 2.5

Reply:
Because the deprecation has changed.